src/indexfile.rs

```diff
@@ -18,25 +18,23 @@ pub struct IndexFile {
 
 impl IndexFile {
     pub fn create(path: &std::path::Path, is_readonly: bool) -> ErrorResult<IndexFile> {
-        let indexfile: std::fs::File;
-
-        if is_readonly {
-            indexfile = OpenOptions::new().read(true).open(&path)?;
+        let indexfile = if is_readonly {
+            OpenOptions::new().read(true).open(&path)?
         } else {
-            indexfile = OpenOptions::new()
+            OpenOptions::new()
                 .read(true)
                 .append(true)
                 .write(true)
                 .create(true)
-                .open(&path)?;
-        }
+                .open(&path)?
+        };
```

Comment:
`if` are expressions in rust, so they return their body.

we can simply just do `let foo = if cond { foo } else { bar };` instead